### DISCUSSION TITLE
Barcode/QR code questions

### DISCUSSION THREAD

**Discussion by luwol03** [inventree/InvenTree#3353] [Q&A] [answered]
Hello,

I find inventree very useful, but I have a few questions related to the bar-/QR-code.

1. Why does this dialog (top navigation bar > scan icon) say scan barcode, but only allows to scan a QR code? Is it also possible to scan barcodes (e.g. the mentioned one from question 2) with this functionallity?
  ![image](https://user-images.githubusercontent.com/60048565/179562926-4f81a523-c12e-4008-a9f8-66d3f6c34f67.png)
2. I read somewhere that its possible to attach an barcode printed on a component to a stock item for easier labeling, but I cannot remember where I read about neither find that functionality in the webui.
3. I saw there are two different ways to generate a QR code for: parts and stock items. Is that correct or is there an third/... typ of QR code? And how do you suggest using they? Should I attach a stock barcode or part barcode to the little ESD boxes or cardboards, ...? When do you suggest which type of code (part, stock item, ...)?

**Comment by matmair** [comment, marked as answer]
1:
This scanner function is set to qrcode and code128 at the moment. There is a textfield below for inputs by keyboard barcode scanners. QR-Codes are often considered a type of 2d-barcodes.
2: https://inventree.readthedocs.io/en/latest/app/stock/#assign-barcode
3: The barcodes are coded for each item so they directly take you to the right actions for that type in the app. Use the barcode that is matching what kind of thing you want to tag. You are missing at least stock location barcodes but the collection might grow in the future.

**Reply by luwol03** [reply]
1. Is it possible to scan other barcodes with this dialog? Or is the problem the underlaying library which is not able to handle other barcodes?
2. thank you~~, but where is that in the webui?~~ -> does tos also only scan barcode128 codes?
3. oh right. Thank you.

**Reply by SchrodingersGat** [reply]
@luwol03 you can assign a custom barcode to a stock item in the web ui: 

![image](https://user-images.githubusercontent.com/10080325/179873136-772b69f9-d787-476b-8de8-af1b342e4a92.png)
